recipe/build-uv-build.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+set -eux
+
+echo "ensuring rust-version in Cargo.toml:#/workspace/package/rust-version is ${CBC_RUST_VERSION}"
+CARGO_TOML_RUST_VERSION=$(grep -iE "rust-version = \".*\"" Cargo.toml)
+
+if [[ "${CARGO_TOML_RUST_VERSION}" == "rust-version = \"${CBC_RUST_VERSION}\"" ]] ;
+then
+  echo "OK rust version in Cargo.toml and conda_build_config.yaml agree: ${CBC_RUST_VERSION}"
+else
+  echo "ERROR rust version unexpcted"
+  echo "... please update recipe/conda_build_config.yaml#/rust_compiler_version"
+  echo "    to match ${CARGO_TOML_RUST_VERSION}"
+  exit 2
+fi
```

Comment:
I don't understand why we need this

Reply:
on more than one occasion on `uv-feedstock`, I've run into issues where the globally-pinned/latest `rust` compiler causes issues. Since adding this, yes, it has failed when it's _wrong_, but it hasn't failed on really jank undefined symbol issues.